spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/InterceptorAutoConfiguration.java

```diff
@@ -0,0 +1,62 @@
+package org.axonframework.springboot.autoconfig;
+
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.gateway.CommandGateway;
+import org.axonframework.config.ConfigurerModule;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.MessageDispatchInterceptor;
+import org.axonframework.messaging.MessageHandlerInterceptor;
+import org.axonframework.queryhandling.QueryBus;
+import org.axonframework.queryhandling.QueryMessage;
+import org.axonframework.spring.config.SpringConfigurer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+
+import java.util.List;
+
+@AutoConfiguration
+@ConditionalOnClass(SpringConfigurer.class)
+@AutoConfigureAfter({
+        AxonAutoConfiguration.class,
+        JpaAutoConfiguration.class,
+        JpaEventStoreAutoConfiguration.class,
+        NoOpTransactionAutoConfiguration.class,
+        TransactionAutoConfiguration.class
+})
+public class InterceptorAutoConfiguration {
+
+    @ConditionalOnClass(MessageDispatchInterceptor.class)
+    @Autowired
+    public void commandDispatchInterceptorConfigurer(CommandGateway commandGateway, List<MessageDispatchInterceptor<? super CommandMessage<?>>> interceptors) {
```

Comment:
I am curious how this behaves if there are two `CommandGateway` instances within the application.
Although this scenario may sound off, I know that there are users who define custom `CommandGateway` beans for different entry points into the system.

So, I think it would be good to add a test case of when this occurs!

Reply:
What would be the intended behaviour for two CommandGateways? I actually asked that myself while implementing  and decided to not iterate over all existing ones to leave it to the developer how to configure a second CommandGateway.

With the @Autowired my assumption is, that always the primary one would be injected.

Reply:
Yea, fair. Two `CommandGateway` instances is something I haven't seen a lot of people do. It becomes more apparent if people use the `CommandGatewayFactory` to generate several interface-specific `CommandGateway` instances.

However, as pointed out, that doesn't happen to often.

I think your assumption that **IF** there are two, one will be primary, and that'll be fetched through the `@Autowired` annotation automatically, should be sufficient.